draft-hzpa-dprive-xfr-over-tls.md

```diff
@@ -148,69 +149,408 @@ DoT: DNS-over-TLS as specified in [@!RFC7858]
+primary to the secondary. A secondary may also initiate an AXFR based on a
+refresh timer or scheduled/triggered zone maintenance.
+
+2. The secondary will normally (but not always) make a SOA query to the primary
+to obtain the serial number of the zone held by the primary.
+
+2. If the primary serial is higher than the secondaries serial, the secondary
+makes an AXFR request (over TCP) to the primary after which the AXFR data flows
+in one or more AXFR responses on the TCP connection.
+
+[@RFC5936] specifies that AXFR must use TCP as the transport protocol but
+details that there is no restriction in the protocol that a single TCP session
+must be used only for a single AXFR exchange, or even solely for XFRs. For
+example, it outlines that the SOA query can also happen on this connection.
+However, this can cause interoperability problems with older implementations
+that support only the trivial case of one AXFR per connection.
+
+TODO: Detail the limitations in existing AXFR implementations as outlined in
+[@RFC5936]
+
+## IXFR Mechanism
+
+The figure below provides an outline of the IXFR mechanism including NOTIFYs.
+
+[Figure 1. IXFR Mechanism](https://docs.google.com/drawings/d/1j0bzoubZvXXt_NnpUJJRoAxM5VjR6OInWPe_jYEql6Y/edit?usp=sharing)
+
+1. An IXFR is normally (but not always) preceded by a NOTIFY (over UDP) from the
+primary to the secondary. A secondary may also initiate an IXFR based on a
+refresh timer or scheduled/triggered zone maintenance.
+
+2. The secondary will normally (but not always) make a SOA query to the primary
+to obtain the serial number of the zone held by the primary.
+
+3. If the primary serial is higher than the secondaries serial, the secondary
+makes an IXFR request (over UDP) to the primary after the primary sends an IXFR
+response.
+
+[@!RFC1995] specifies that Incremental Transfer may use UDP if the entire IXFR
+response can be contained in a single DNS packet, otherwise, TCP is used. In
+fact is says in non-normative language:
+
+"Thus, a client should first make an IXFR query using UDP."
+
+So there may be a forth step above where the client falls back to IXFR-over-TCP.
+There may also be a forth step where the secondary must fall back to AXFR
+because e.g. the primary does not support IXFR.
+
+## Data Leakage of NOTIFY and SOA Message Exchanges
+
+This section attempts to presents a rationale for also encrypting the other
+messages in the XFR mechanism.
+
+Since the SOA of the zone can be trivially discovered by simply querying the
+authoritative server leakage RR of this is not discussed in the following
+sections.
+
+### NOTIFY
+
+Unencrypted NOTIFY messages identify configured secondaries on the primary.
+
+[@RFC1996] also states: 
+
+"If ANCOUNT>0, then the answer section represents an
+  unsecure hint at the new RRset for this <QNAME,QCLASS,QTYPE>.
+  
+But since the only supported QTYPE for NOTIFY is SOA, this does not pose a
+potential leak.
+
+### SOA
+
+QUESTION: No real leakage here?
```

Comment:
If you see a NOTIFY followed by a SOA query, it is leaking that the secondary has an out-of-date zone...
This might be revealed from just the notify too, assuming that the secondary is publicly query-able.